src/client/Microsoft.Identity.Client.Broker/WamAdapters.cs

```diff
@@ -240,5 +239,15 @@ private static string GetExpectedRedirectUri(string clientId)
         {
             return $"ms-appx-web://microsoft.aad.brokerplugin/{clientId}";
         }
+
+        /// <summary>
+        /// Get ExpiresIn from Auth Result
+        /// </summary>
+        /// <param name="expiresOn"></param>
+        private static long GetExpiresInFromAuthResult(DateTime expiresOn)
+        {
+            return DateTimeHelpers.GetDurationFromNowInSeconds
```

Comment:
This seems very inefficient to transform a DateTime into Unix timestamp. The `expiresOn` can be assumed to be UTC. Can you find a better way to transform a DateTime into what MSAL needs?

Reply:
updated